edb/pgsql/resolver/relation.py

```diff
@@ -385,16 +390,26 @@ def _construct_column(p: s_pointers.Pointer, ctx: Context) -> context.Column:
         elif p.is_id_pointer(ctx.schema):
             col.reference_as = 'id'
         else:
-            _, dbname = pgcommon.get_backend_name(
-                ctx.schema, p, catenate=False
-            )
+            _, dbname = pgcommon.get_backend_name(ctx.schema, p, catenate=False)
             col.reference_as = dbname
 
-    if isinstance(p, s_links.Link):
-        col.name = p.get_shortname(ctx.schema).name + '_id'
-
-        _, dbname = pgcommon.get_backend_name(ctx.schema, p, catenate=False)
-        col.reference_as = dbname
+    elif isinstance(p, s_links.Link):
+        if short_name.name == '__type__':
+            col.name = '__type__'
+            col.reference_as = '__type__'
```

Comment:
Ugh, should we call it `__type__id`? That seems awful though

Reply:
It does look awful. It's a special-case column anyway, so we can change it into `the_id_of_the_type_of_this_object`, but I do think we can just leave it as `__type__`.